src/App.tsx

```diff
@@ -1,61 +1,127 @@
 import './App.scss';
+import React, { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './Types/User';
+import { Todo } from './Types/Todo';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUser(userId: number): User | null {
+  const currentUser = usersFromServer.find(user => user.id === userId);
+
+  return currentUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [selectedUser, setSelectedUser] = useState(false);
+  const [enteredTitle, setEnteredTitle] = useState(false);
+
+  const newId = Math.max(...todos.map(todo => todo.id)) + 1;
+
+  const addTodo = (id: number) => {
+    if (getUser(id) && title.trim()) {
+      todos.push({
```

Comment:
store todos in state. And it is really bad idea to modificate array & objects, do their copy instead